src/sr_entities_handler.erl

```diff
@@ -70,7 +70,19 @@ resource_exists(Req, State) ->
 -spec content_types_accepted(cowboy_req:req(), state()) ->
   {[{{binary(), binary(), '*'}, atom()}], cowboy_req:req(), state()}.
 content_types_accepted(Req, State) ->
-  {[{{<<"application">>, <<"json">>, '*'}, handle_post}], Req, State}.
+    #{opts := #{path := Path}} = State,
+    {Method, Req2} = cowboy_req:method(Req),
+    try
+        #{metadata := Metadata} = trails:retrieve(Path),
+        AtomMethod = method_to_atom(Method),
+        #{AtomMethod := #{consumes := Consumes}} = Metadata,
+        Handler = compose_handler_name(AtomMethod),
+        RetList = [{list_to_binary(X), Handler} || X <- Consumes],
+        {RetList, Req2, State}
+    catch
+        _:_ ->
+            {[{<<"application/json">>, handle_post}], Req, State}
```

Comment:
This is not good, by default we want to allow `"application/json"` but also `"application/json; charset=utf8"`. Thus the tuple with `'*'`.